ss_wave_2.rb

```diff
@@ -0,0 +1,219 @@
+
+  if solar_system.planet_list.include?(planet_choice)
+    return solar_system.pick_planet(planet_choice)
+  else
+    while solar_system.planet_list.include?(planet_choice) == false
+      print "Please enter a planet on the list.\s"
+      planet_choice = gets.chomp.downcase.capitalize
+      return solar_system.pick_planet(planet_choice)
+
+    end
+
+  end
+end
+
+
+def know_another_question
+  puts "Would you like to know more about another planet? Enter (Y/N)\s"
+
+end
+
+def which_planet_question
+  puts "which planet would you like to know more about?"
+
+end
+
+def create_planet_question
+  puts "Would you like to add a planet?"
+
+end
+def user_planet_name
+  puts "what would you like to name your planet?"
+
+end
+
+
+# defines and adds a new planet to the SolarSystem
+def add_planet(name, date_discovered,input)
+  planet = Planet.new(name, date_discovered, input)
+  return planet
+
+end
+
+# User interface
+puts "Welcome to the Solar System App!"
+puts "Click around and see what you can find!"
+
+puts my_ss.planet_list
+
+# Checks to see if user wants to know more about the planets in the solar system
+print "Would you like to know more about a particular planet? Enter (Y/N)\s"
+know_more = gets.chomp.downcase
+# If the user wants to see more about the solar system check to see which planet
+if know_more == "y" || know_more == "yes"
+  which_planet_question()
+  which_planet = gets.chomp
+  user_choice(my_ss,which_planet)
+  know_another_question()
+  know_another = gets.chomp.downcase
+  while know_another == "y" || know_another == "yes"
+    which_planet_question()
+    which_planet = gets.chomp
+    user_choice(my_ss, know_another)
+    know_another_question
+    know_another = gets.chomp.downcase
+  end
+  case know_another
+  when "n","no"
+    create_planet_question
+    user_create = gets.chomp.downcase
+    if user_create == "y" || user_create == "yes"
+      user_planet_name
+      user_planet = gets.chomp
+      puts "when was your planet discovered?"
+      user_planet_details = gets.chomp
+      puts "how many attributes does your planet have?"
+      attributes = gets.chomp
+      add_planet(user_planet, user_planet_details, attributes)
+      # puts here to test
+      puts "what attribute would you like to add?(i.e. gravity, mass, length of a day.)"
+      user_key = gets.chomp
+      puts "what is the value of #{user_key}? (i.e. mass is 317.8)"
+      user_value = gets.chomp
+      change_value(user_key, user_value)
+
+      planet.planet_attributes
+
+      my_ss.add_planet(planet)
+
+
+      puts "You have added a new planet to the solar system!"
+    else
+      puts "Thank you for using the Solar System App!"
+    end
+
+  end
+elsif know_more == "n" || know_more == "no"
+  create_planet_question
+  user_create = gets.chomp.downcase
+  if user_create == "y" || user_create == "yes"
+    user_planet_name
+    user_planet = gets.chomp
+    puts "when was your planet discovered?"
```

Comment:
This code is almost exactly the same as the planet creation code you've got starting on line 166. Could you consolidate the two somehow?